node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -988,4 +993,81 @@ mod tests {
         let expected = ExternalData::new(DEFAULT_CHAIN, NeighborhoodModeLight::ZeroHop, None);
         assert_eq!(result, expected)
     }
+
+    fn check_data_directory_combinations(
+        chain_opt: Option<&str>,
+        data_directory_opt: Option<&str>,
+        expected: Option<&str>,
+    ) {
+        let home_dir = PathBuf::from("/home/cooga");
+        let standard_data_dir = PathBuf::from("/home/cooga/.local");
+
+        let args = match (chain_opt, data_directory_opt) {
+            (Some(chain_opt), Some(data_directory_opt)) => ArgsBuilder::new()
+                .param("--chain", chain_opt)
+                .param("--real-user", "999:999:/home/cooga")
+                .param("--data-directory", data_directory_opt),
+            (None, Some(data_directory_opt)) => ArgsBuilder::new()
+                .param("--chain", "polygon-mainnet")
+                .param("--data-directory", data_directory_opt)
+                .param("--real-user", "999:999:/home/cooga"),
+            (Some(chain_opt), None) => ArgsBuilder::new()
+                .param("--chain", chain_opt)
+                .param("--real-user", "999:999:/home/cooga"),
+            (None, None) => ArgsBuilder::new()
+                .param("--chain", "polygon-mainnet")
```

Comment:
See above about `chain` passed as `None` but specified on the command line anyway.

Reply:
I did not found any other way to pass existing setup to `server_initializer_collected_params` function.